Sming/Core/Network/Http/HttpServerConnection.cpp

```diff
@@ -139,15 +144,25 @@ int HttpServerConnection::onHeadersComplete(const HttpHeaders& headers)
 
 int HttpServerConnection::onBody(const char* at, size_t length)
 {
-	if(bodyParser) {
-		size_t consumed = bodyParser(request, at, length);
-		if(consumed != length) {
-			return -1;
+	if(!bodyContentError) {
```

Comment:
Re-arrange a bit the if statements to avoid too many nested ifs and improve readability. For example that block here can be rewritten as:
```
     if(bodyContentError) {
            return 0;
     }

    if(bodyParser) {
          // ...
    }

    if(resource != nullptr && resource->onBody) {
       // ...
    }

    return 0;
```

Reply:
done